irohad/model/impl/query_execution.cpp

```diff
@@ -313,17 +313,21 @@ QueryProcessingFactory::executeGetAccountTransactions(
 
 QueryProcessingFactory::QueryResponseBuilderDone
 iroha::model::QueryProcessingFactory::executeGetTransactions(
-    const shared_model::interface::GetTransactions &query) {
-  const std::vector<shared_model::crypto::Hash> &hashes =
-      query.transactionHashes();
+    const shared_model::interface::GetTransactions &q,
+    const shared_model::interface::types::AccountIdType &accountId) {
+  const std::vector<shared_model::crypto::Hash> &hashes = q.transactionHashes();
 
   auto transactions = _blockQuery->getTransactions(hashes);
 
   std::vector<shared_model::proto::Transaction> txs;
+  bool can_get_all =
+      checkAccountRolePermission(accountId, *_wsvQuery, can_get_all_txs);
   transactions.subscribe([&](const auto &tx) {
     if (tx) {
-      txs.push_back(
-          *std::static_pointer_cast<shared_model::proto::Transaction>(*tx));
+      auto proto_tx =
+          *std::static_pointer_cast<shared_model::proto::Transaction>(*tx);
+      if (can_get_all or proto_tx.creatorAccountId() == accountId)
+        txs.push_back(proto_tx);
```

Comment:
Still concerned about that line, ping @lebdron

Reply:
`proto_tx` is pushed with lvalue, so copy ctor is called, looks fine to me.